### PR TITLE
Fixed issues due gradle 4.10 and kotlin-dsl 1.0-rc

### PR DESCRIPTION
**Pull request by mstrangfeld**
I implemented the new API and removed some deprecated methods.
This will bump the minimum Gradle version up to 4.10

### CODE REVIEW

File: src/test/kotlin/testsupport/gradleTestTemplate.kt [hunk 1]
```diff
@@ -91,7 +88,7 @@ private data class GradleProjectInvocationContext(
 ) : TestTemplateInvocationContext {
 
   override fun getDisplayName(invocationIndex: Int): String  = "[Gradle " +
-    if (version == GradleVersion.current()) {
```
Comment (mkobit) [inline src/test/kotlin/testsupport/gradleTestTemplate.kt]: `==` checks for structural equality and gets compiled down to (basically) `.equals` anyways (see https://kotlinlang.org/docs/reference/equality.html#structural-equality). I think `==` is the right kotlin style, so i'd like to stick with it unless im missing something

would you mind changing it back to `==`?

File: src/test/kotlin/testsupport/gradleTestTemplate.kt [hunk 2]
```diff
@@ -121,7 +118,7 @@ private class FilteringGradleExecutionCondition(
     return when (propertyValue) {
       "all", "default" -> ConditionEvaluationResult.enabled("All tests enabled through $propertyValue filter")
       "current" -> {
-        if (targetVersion == GradleVersion.current()) {
```
Comment (mkobit) [inline src/test/kotlin/testsupport/gradleTestTemplate.kt]: same comment here about `==`

**Comment by mkobit** [comment]
I can take care of updating the Circle CI configuration (at https://github.com/mkobit/jenkins-pipeline-shared-libraries-gradle-plugin/blob/efb25c9aaba86a650d21e128703f02e510a2a49a/.circleci/config.yml). I don't know why it isn't being triggered here, but that is a different issue.

File: src/test/kotlin/testsupport/gradleTestTemplate.kt (hunk 1 above)
Comment (mstrangfeld) [reply src/test/kotlin/testsupport/gradleTestTemplate.kt]: When I use `==` the compiler fails:
`No method 'equals(Any?): Boolean' available`
But I honestly have no clue why. First I thought it was a null safety problem but even with `version!! == GradleVersion.current()` it is the same problem...

Comment (mkobit) [reply src/test/kotlin/testsupport/gradleTestTemplate.kt]: ok strange - I'll merge and then figure it out

**Comment by mkobit** [comment]
Verified locally, looks like I need to take another pass at JVM OOMs on Circle CI :disappointed:

**Pull request merged**